src/coreclr/tests/src/JIT/Regression/JitBlue/GitHub_36614/GitHub_36614.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace projs
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            Program p = new Program();
+            p.NarrowDouble();
+            Console.WriteLine("Passed.");
+            return 100;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void NarrowDouble()
+        {
+            // GenerateSource1 and GenerateSource2 methods are needed to exercise the bug code path.
+            double[] source1 = GenerateSource1();
+            double[] source2 = GenerateSource2();
+            Vector<double> sourceVec1 = new Vector<double>(source1);
+            Vector<double> sourceVec2 = new Vector<double>(source2);
+            Vector<float> dest = Vector.Narrow(sourceVec1, sourceVec2);
+
+            for (int i = 0; i < Vector<double>.Count; i++)
+            {
+                Assert.Equal(unchecked((float)source1[i]), dest[i]);
+            }
+            for (int i = 0; i < Vector<double>.Count; i++)
+            {
+                // The test represents a problem where stack level is not adjusted in the offset while trying to
```

Comment:
You might want a comment here that points out this only affects x86, when double-alignment causes access to a stack local.